python/ray/serve/tests/test_deploy.py

```diff
@@ -389,14 +388,22 @@ def test_redeploy_multiple_replicas(serve_instance, use_handle):
     def call(block=False):
         if use_handle:
             handle = serve.get_deployment(name).get_handle()
-            ret = ray.get(handle.handler.remote(block))
+            ret = ray.get(handle.handler.remote(block)).split("|")
         else:
             ret = requests.get(
-                f"http://localhost:8000/{name}", params={
-                    "block": block
-                }).text
+                f"http://localhost:8000/{name}", params={"block": block})
 
-        return ret.split("|")[0], ret.split("|")[1]
+            ret_candidate = ret.text.split("|")
+
+            # Set return value to (None, None)
+            # if status_code is 404 or if there are
+            # not enough values to return
+            if ret.status_code == 404 or len(ret_candidate) != 2:
+                ret = (None, None)
+            else:
+                ret = ret_candidate
+
+        return ret[0], ret[1]
```

Comment:
Let me know if this approach looks good. I will apply this to other test cases as well.

Reply:
This approach looks good to me. cc @simon-mo